src/witty/compile_module.py

```diff
@@ -122,8 +380,17 @@ def compile_module(
         force_rebuild = os.getenv("WITTY_FORCE_REBUILD", "0").lower() in ("1", "true")
 
     module_hash = _generate_hash(
-        source_pyx, source_files, extra_compile_args, extra_link_args, extension_kwargs
+        source, source_files, extra_compile_args, extra_link_args, extension_kwargs
     )
+
+    if source_type == "nanobind":
+        assert name is None, (
+            "For non-PYX builds, the module name should not be given but be part "
+            "of the source file."
+        )
+        name = re.findall(r"NB_MODULE\s*\(\s*(.*),", source)
+        assert len(name) == 1, "More than one NB_MODULE found, only one allowed."
```

Comment:
```suggestion
        names = re.findall(r"NB_MODULE\s*\(\s*(.*),", source)
        if len(names) != 1:
            raise ValueError(
                "For nanobind modules, the source must contain a single NB_MODULE "
                f"definition with the module name.  Found {len(names)}"
            )
```

for typing... using temporary `names` var, and also checking len 0